server/src/main/java/com/google/tsunami/security/scanner/utilities/TsunamiTestbedUtil.java

```diff
@@ -0,0 +1,159 @@
+        System.out.println("Waiting for service to start...");
+        Thread.sleep(1000);
+      }
+
+      // Get Unique Id of certain application.
+      V1ServiceList v1ServiceList =
+          coreV1Api.listNamespacedService(
+              "default", null, null, null, null, null, null, null, null, null);
+
+      for (V1Service svc : v1ServiceList.getItems()) {
+        if (svc.getMetadata().getName().equals(application)) {
+          jobId = svc.getMetadata().getUid();
+          System.out.println("Application " + application + "'s unique id is: " + jobId);
+        }
+      }
+
+      return jobId;
+    }
+  }
+
+  public List<String> listApplications() throws ApiException {
+    synchronized (lock) {
+      List<String> applications = new ArrayList<>();
+
+      System.out.println("List running applications.");
+
+      CoreV1Api coreV1Api = new CoreV1Api();
+
+      // List all running services.
+      V1ServiceList v1ServiceList =
+          coreV1Api.listNamespacedService(
+              "default", null, null, null, null, null, null, null, null, null);
+      for (V1Service svc : v1ServiceList.getItems()) {
+        applications.add(svc.getMetadata().getName());
+      }
+
+      System.out.println("Applications list: " + applications);
+
+      return ImmutableList.copyOf(applications);
+    }
+  }
+
+  public ServiceEndpoint getServiceEndpoint(String application) throws ApiException {
+    synchronized (lock) {
+      // Set default ip and port.
+      ServiceEndpoint serviceEndpoint =
+          ServiceEndpoint.newBuilder().setIp("00.00.00.00").setPort("80").build();
+
+      System.out.println("Getting ip and port information of : " + application);
+
+      // Get services list.
+      CoreV1Api coreV1Api = new CoreV1Api();
+      V1ServiceList v1ServiceList =
+          coreV1Api.listNamespacedService(
+              "default", null, null, null, null, null, null, null, null, null);
+
+      for (V1Service svc : v1ServiceList.getItems()) {
+        // Find the required service.
+        if (svc.getMetadata().getName().equals(application)) {
+
+          // Get service's ip.
+          try {
+            List<V1LoadBalancerIngress> svcIP = svc.getStatus().getLoadBalancer().getIngress();
+            for (V1LoadBalancerIngress ip : svcIP) {
+              System.out.println("Application " + application + "'s IP: " + ip.getIp());
+              serviceEndpoint.toBuilder().setIp(ip.getIp()).build();
+            }
+          } catch (NullPointerException e) {
+            System.out.println("Caught null pointer exception.");
+          }
+
+          // Get service's port.
+          List<V1ServicePort> ports = svc.getSpec().getPorts();
+          for (V1ServicePort port : ports) {
+            System.out.println("Application " + application + "'s port: " + port.getPort());
+            serviceEndpoint.toBuilder().setPort(port.getPort().toString()).build();
+          }
+        }
+      }
+
+      return serviceEndpoint;
```

Comment:
Instead of returning a default value when the service cannot be found, throw an exception and let the server to handle the failure case.